packages/flutter/lib/src/painting/continuous_rectangle_border.dart

```diff
@@ -3,58 +3,276 @@
 // found in the LICENSE file.
 
 import 'dart:math' as math;
+import 'dart:ui' as ui;
 
 import 'basic_types.dart';
-import 'border_radius.dart';
 import 'borders.dart';
 import 'edge_insets.dart';
 
-/// A rectangular border with smooth continuous transitions between the straight
-/// sides and the rounded corners.
+/// A rectangle border with continuous corners.
+///
+/// A shape similar to a rounded rectangle, but with a smoother transition from
+/// the sides to the rounded corners.
+///
+/// The rendered shape roughly approximates that of a superellipse. In this
+/// shape, the curvature of each 90º corner around the length of the arc is
+/// approximately a gaussian curve instead of a step function as with a
+/// traditional quarter circle round. The rendered rectangle is roughly a
+/// superellipse with an n value of 5.
+///
+/// In an attempt to keep the shape of the rectangle the same regardless of its
+/// dimension (and to avoid clipping of the shape), the radius will
+/// automatically be lessened if its width or height is less than ~3x the
+/// declared radius. The new resulting radius will always be maximal in respect
+/// to the dimensions of the given rectangle.
+///
+/// The ~3 represents twice the ratio (ie. ~3/2) of a corner's declared radius
+/// and the actual height and width of pixels that are manipulated to render it.
```

Comment:
> manipulated to render it

This explanation is a bit reversed. You're describing the consequence but you want to describe the cause, i.e. more space is needed to gradually smooth out the curve.